src/main/java/randoop/condition/ExpectedOutcomeTable.java

```diff
@@ -0,0 +1,156 @@
+ * declarations that it overrides or implements.
+ *
+ * <ol>
+ *   <li>Whether any guard expression for the {@link randoop.condition.specification.Precondition}
+ *       fails, or all are satisfied.
+ *   <li>The set of {@link ThrowsCondition} objects for expected exceptions. An exception is
+ *       expected if the guard of a {@link GuardThrowsPair} is satisfied.
+ *   <li>The expected {@link PropertyExpression}, if any.
+ * </ol>
+ *
+ * <p>For a particular operation call, a table is constructed by calling {@link
+ * OperationConditions#checkPrestate(Object[])} and represents the expected outcome(s) from the
+ * call. The table is used to create a {@link TestCheckGenerator} by calling {@link
+ * #addPostCheckGenerator(TestCheckGenerator)} that, when given to the sequence generator, will
+ * classify the call as follows
+ *
+ * <ol>
+ *   <li>For each table entry with a non-empty expected exception set
+ *       <ul>
+ *         <li>If an exception is thrown by the call and the thrown exception is a member of the
+ *             set, then classify as {@link randoop.main.GenInputsAbstract.BehaviorType#EXPECTED}.
+ *         <li>If an exception is thrown by the call and the thrown exception is not a member of the
+ *             set, classify as {@link randoop.main.GenInputsAbstract.BehaviorType#ERROR} (because
+ *             the specification required an exception to be thrown, but it was not thrown).
+ *         <li>If no exception is thrown, then classify as {@link
+ *             randoop.main.GenInputsAbstract.BehaviorType#ERROR}.
+ *       </ul>
+ *
+ *   <li>If for each table entry, the preconditions failed, classify as {@link
+ *       randoop.main.GenInputsAbstract.BehaviorType#INVALID}.
+ *   <li>For each table entry where all preconditions were satisfied, check the corresponding normal
+ *       post-condition property, if one exists. If any such property fails, then classify as {@link
+ *       randoop.main.GenInputsAbstract.BehaviorType#ERROR}.
+ * </ol>
+ */
+public class ExpectedOutcomeTable {
+
+  /** Indicates whether this table is empty. */
+  private boolean isEmpty = true;
+
+  /** Indicates whether a precondition was satisfied. */
+  private boolean hasSatisfiedGuardExpression = false;
+
+  /** The list of post-conditions whose guard expression was satisfied. */
+  private final List<PropertyExpression> postConditions;
+
+  /** The list of sets of throws clauses for which the guard expression was satisfied. */
+  private final List<Set<ThrowsClause>> exceptionSets;
+
+  /** Creates an empty {@link ExpectedOutcomeTable}. */
+  public ExpectedOutcomeTable() {
+    postConditions = new ArrayList<>();
+    exceptionSets = new ArrayList<>();
+  }
+
+  /**
+   * Adds the outcome of checking the prestate parts of an operation's specification.
+   *
+   * @param guardIsSatisfied boolean value indicating whether all guard expressions are satisfied
```

Comment:
"All guard expressions" is not defined.  Is it only for preconditions, or all guards including for postconditions and throws-conditions?